core-bundle/src/EventListener/DataContainer/RelCanonicalListener.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener\DataContainer;
+
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\CoreBundle\ServiceAnnotation\Callback;
+use Contao\Image;
+use Contao\StringUtil;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ * @internal
+ */
+class RelCanonicalListener
+{
+    private ContaoFramework $framework;
+    private TranslatorInterface $translator;
+
+    public function __construct(ContaoFramework $framework, TranslatorInterface $translator)
+    {
+        $this->framework = $framework;
+        $this->translator = $translator;
+    }
+
+    /**
+     * @Callback(table="tl_page", target="config.onload")
+     */
+    public function disableRoutingFields(): void
+    {
+        $adapter = $this->framework->getAdapter(Image::class);
+
+        $renderHelpIcon = fn () => $adapter->getHtml(
+            'show.svg',
+            '',
+            sprintf(
+                'title="%s"',
+                StringUtil::specialchars($this->translator->trans('tl_page.relCanonical', [], 'contao_tl_page'))
+            )
+        );
+
+        foreach (['canonicalLink', 'canonicalKeepParams'] as $field) {
+            $GLOBALS['TL_DCA']['tl_page']['fields'][$field]['eval']['disabled'] = true;
```

Comment:
Does this field now get always diesabled?
And isn’t this handled already by https://github.com/contao/contao/blob/0ed2c1e25c49d8d6557201423e2be6976ba11ba9/core-bundle/src/EventListener/DataContainer/DisableCanonicalFieldsListener.php#L42

Reply:
Yeah we already have a `DisableCanonicalFieldsListener`, I'd prefer having the logic in the same one (even if that means renaming it but 2 separate seems strange).

Reply:
Please review again.